onmt/model_builder.py

```diff
@@ -232,6 +232,36 @@ def fix_key(s):
                                for k, v in checkpoint['model'].items()}
         # end of patch for backward compatibility
 
+        if model_opt.update_vocab:
+            # Update vocabualry embeddings with checkpoint embeddings
+            logger.info("Updating vocabulary embeddings with checkpoint embeddings")
+            # Embedding layers
+            enc_emb_name = "encoder.embeddings.make_embedding.emb_luts.0.weight"
+            dec_emb_name = "decoder.embeddings.make_embedding.emb_luts.0.weight"
+
+            tgt_multifield = fields["tgt"]
+            checkpoint_tgt_multifield = checkpoint["vocab"]["tgt"]
+            for field_name, emb_name in [("src", enc_emb_name), ("tgt", dec_emb_name)]:
+                multifield = fields[field_name]
+                checkpoint_multifield = checkpoint["vocab"][field_name]
+                for (name, field), (checkpoint_name, checkpoint_field) in zip(multifield, checkpoint_multifield):
```

Comment:
multifield was designed for multiple source fields, there will be old models using the source features. As we've dropped that feature for now, maybe it's more appropriate to directly call `base_field` rather than iterate since the code doesn't handle updating other fields.